adr/002-Custom-Resource-Deserialization-Problem.md

```diff
@@ -0,0 +1,37 @@
+# Multi Version Custom Resources Deserialization Problem
+
+## Status
+
+accepted
+
+## Context
+
+In case there are multiple versions of a custom resource in can happen that a controller/informer tracking
+such a resource might run into deserialization problem as shown
+in [this integration test](https://github.com/java-operator-sdk/java-operator-sdk/blob/07aab1a9914d865364d7236e496ef9ba5b50699e/operator-framework/src/test/java/io/javaoperatorsdk/operator/MultiVersionCRDIT.java#L55-L55)
+. In the mentioned case two versions of a custom resource are not compatible with each other. The informer receives
+both, but naturally not able to deserialize one of them.
+
+How should the framework or the underlying informer behave?
+
+Alternatives:
+
+1. The informer should skip the resource and should continue to process the resources with the correct version.
```

Comment:
IMHO I think this shouldn't even be considered as an alternative. It might encourage users to ignore errors that in the long-term will definitely have an impact in their application's lifecycle.

Reply:
Just adding since it was working this way if until now, and before was mentioned in other conversations. But completely agree with what you say.